Refactored `warn` and `warnAndFix` in `Warnings`

nulls opened the pull request:
### What's done:
- removed `isFixMode` from `warn`
- removed `canBeAutoCorrected` from `warnAndFix`

It closes #1748

kgevorkyan commented on diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/constants/Warnings.kt:
```diff
@@ -264,7 +299,7 @@ enum class Warnings(
     companion object {
         val names by lazy {
-            values().map { it.name }
+            entries.map { it.name }
```
is not the entries the `key` `value` pairs?

nulls replied:
kotlin 1.9 suggests to migrate to `entries` since `values()` will be removed

nulls replied:
and `entries` is full replacement for values() -- same semantic

kgevorkyan reviewed (approved):
I see that you change in some rules argument `canBeAutocorrected` from `false` to `true`
could you please update corresponding .md files? `avaliable-rules` and so on

nulls commented:
> I see that you change in some rules argument `canBeAutocorrected` from `true` to `false` could you please update corresponding .md files? `avaliable-rules` and so on

changed only `AVOID_NULL_CHECKS` in `Warnings`. Updated `rules-mapping.md` and `available-rules.md` (there already was marked as fix, updated notes only)
